Zilliqa/contracts/xsgd_contract.scilla

```diff
@@ -77,12 +77,38 @@ let three_msgs =
   fun (msg3 : Message) =>
     let msgs_tmp = two_msgs msg2 msg3 in
     Cons {Message} msg1 msgs_tmp
-let get_minter_allowance =
-  fun (option_allowance : Option (Option Uint128)) =>
-    match option_allowance with
-    | Some (Some allowance) => allowance
+    
+let deconstruct_option_uint128 =
+  fun (option_uint128 : Option Uint128) =>
+    match option_uint128 with
+    | Some a => a
+    | _ => zero
+    end
+
+let deconstruct_option_option_uint128 =
+  fun (option_option_uint128 : Option (Option Uint128)) =>
+    match option_option_uint128 with
+    | Some (Some a) => a
     | _ => zero
     end
+    
+(* Only use this for DecreaseAllowance and DecreaseMinterAllowance *)
+let better_subtract = 
```

Comment:
`safe_subtract`? 😂

Reply:
Hmm the math in scilla is already considered safe tho, like it automatically reverts if there's an under/overflow so a transition cant be called "safe subtract" when it provides this functionality natively within Scilla liao

But ya, i'm flexible to change XD